hazelcast/src/main/java/com/hazelcast/cache/impl/nearcache/impl/record/AbstractNearCacheRecord.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2008-2013, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl.nearcache.impl.record;
+
+import com.hazelcast.cache.impl.nearcache.NearCacheRecord;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Abstract implementation of {@link NearCacheRecord} with value and
+ * expiration time as internal state.
+ *
+ * @param <V> the type of the value stored by this {@link AbstractNearCacheRecord}
+ */
+public abstract class AbstractNearCacheRecord<V> implements NearCacheRecord<V> {
+
+    protected V value;
+    protected long creationTime = TIME_NOT_SET;
```

Comment:
`creationTime` is also set by constructor with passed argument. So I didn't make it `final`